### PR TITLE
docs: improve crate documentation

### LINKED ISSUE #40: Provide package documentation
Should be built with `cargo doc`.

### PR DESCRIPTION
**Pull request by saidsay-so**
Closes #40

### CODE REVIEW

File: book/src/compatibility.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  - ZFS
+- Linux
+  - ext4
+
+### Experimental support
+
+The test suite has not been tested on the following operating systems, but should work:
+
+- macOS
+- netBSD
```
Comment (asomers) [inline book/src/compatibility.md:24]:
```suggestion
- MacOS
- NetBSD
```

File: rust/src/config.rs [hunk 2]
```diff
@@ -28,8 +40,10 @@ pub struct FeaturesConfig {
 /// Please see the book for more details.
 #[derive(Debug, Serialize, Deserialize)]
 pub struct SettingsConfig {
+    /// Time to sleep between tests.
```
Comment (asomers) [inline rust/src/config.rs:43]: This line isn't accurate.  `naptime` isn't used between tests; it's used within tests.  The requirement is that it should be greater than the timestamp granularity of the file system under test.

File: rust/src/macros.rs [hunk 3]
```diff
@@ -1,20 +1,46 @@
-#[macro_export]
+//! Macros for defining test cases.
+//! 
+//! The `test_case` macro is used to define test cases.
+
+/// The `test_case` macro is used to define test cases for a test suite.
+/// 
+/// The macro is used to define a test case for a test suite.
```
Comment (asomers) [inline rust/src/macros.rs:7]: These two sentences are redundant.

**Review by asomers — CHANGES_REQUESTED** [review]
This looks like a good change.  Is there a rendered version of the book online somewhere where I may see it?

File: rust/src/config.rs (hunk 2 above)
Comment (saidsay-so) [reply rust/src/config.rs]: Fixed

File: rust/src/config.rs [hunk 4]
```diff
@@ -40,10 +41,14 @@ pub struct FeaturesConfig {
 /// Please see the book for more details.
 #[derive(Debug, Serialize, Deserialize)]
 pub struct SettingsConfig {
-    /// Time to sleep between tests.
+    /// Time to sleep within tests (in milliseconds)
```
Comment (asomers) [inline rust/src/config.rs:44]:
```suggestion
    /// Time to sleep within tests (in seconds)
```

**Review by asomers — CHANGES_REQUESTED** [review]

**Comment by saidsay-so** [comment]
> This looks like a good change. Is there a rendered version of the book online somewhere where I may see it?

Unfortunately, you have to build it to preview it. You have to run `./build.sh` in the `book` directory.

**Review by asomers — APPROVED** [review]

**Pull request merged**